lib/fun_with_flags/application.ex

```diff
@@ -42,7 +41,21 @@ defmodule FunWithFlags.Application do
   #
   defp notifications_spec do
     try do
-      Config.change_notifications_enabled? && Config.notifications_adapter.worker_spec()
+      if Config.change_notifications_enabled? do
+
+        {driver, opts} = Config.notifications_pubsub_driver() |> case do
+          mod when is_atom(mod) -> {mod, []}
+          {mod, opts} = x when is_atom(mod) and is_list(opts) -> x
+        end
```

Comment:
I find this a bit difficult to read, how about:
```suggestion
        {driver, opts} =
          case Config.notifications_pubsub_driver() do
            mod when is_atom(mod) ->
              {mod, []}
            {mod, opts} when is_atom(mod) and is_list(opts) ->
              {mod, opts}
            _ ->
              raise "unexpected config for ...."
          end
```